DSP_Plugins.GalacticScale/Scripts/PatchPlanetSize/PatchNearColliderLogic.cs

```diff
@@ -0,0 +1,161 @@
+            {
+                int n = PlanetPhysics.HashPhysBlock(pos);
+                if (n == -1)
+                    break;
+                for (int index = 0; index < ___activeColHashCount; ++index)
+                {
+                    if (___activeColHashes[index] == n)
+                        break;
+                }
+                ___activeColHashes[___activeColHashCount++] = n;
+            }
+            if (___activeColHashCount > 0) //If we have some hashes
+            {
+                for (int index1 = 0; index1 < ___activeColHashCount; ++index1) 
+                {
+                    int activeColHash = ___activeColHashes[index1]; //grab it's value (445 etc)
+                    ColliderData[] colliderPool = ___colChunks[activeColHash].colliderPool; //and Grab the Collider Pool from the collider container ___colChunks[that hash]
+                    for (int index2 = 1; index2 < ___colChunks[activeColHash].cursor; ++index2) //For values between 1 and some value called cursor from that collider container, which seems to be between 1 and 10...
+                    {
+                        if (colliderPool[index2].idType != 0 && //check array items 1-10 from the colliderPool and see if their idtype isnt 0
+                            colliderPool[index2].usage != EColliderUsage.Build &&  //that they are not a building
+                            colliderPool[index2].objType == EObjectType.Vein &&  //that their objType is a vein
+                            (double)(colliderPool[index2].pos - center).sqrMagnitude <= (double)areaRadius * (double)areaRadius + (double)colliderPool[index2].ext.sqrMagnitude) //and their center is within the area we are scanning
+                        {
+                            bool flag = false; //if so, dont set the break flag
+                            for (int index3 = 0; index3 < num; ++index3) //hang on a second, these are while loops :S
+                            {
+                                if (veinIds[index3] == colliderPool[index2].objId)
+                                {
+                                    flag = true;
+                                    break;
+                                }
+                            }
+                            if (!flag)
+                                veinIds[num++] = colliderPool[index2].objId;
+                        }
+                    }
+                }
+            }
+            __result = num;
+            return false;
+        }
+
+        [HarmonyPrefix]
+        [HarmonyPatch("GetVegetablesInAreaNonAlloc")]
+        public static bool GetVegetablesInAreaNonAlloc(ref int __result, Vector3 center, float areaRadius, int[] vegeIds, ref int ___activeColHashCount, ref int[] ___activeColHashes, ref ColliderContainer[] ___colChunks)
+        {
```

Comment:
Same thing : Comment / details the changes you made to the function

Reply:
I did, there were only two lines